test/e2e/reconciler/clusterworkspace/controller_test.go

```diff
@@ -229,6 +222,13 @@ func toYAML(t *testing.T, obj interface{}) string {
 	return string(bs)
 }
 
+func getKcpClusterClient(t *testing.T, config *rest.Config, clusterName logicalcluster.Name) kcpclientset.Interface {
+	clusterCfg := kcpclienthelper.ConfigWithCluster(config, clusterName)
+	kcpClusterClient, err := kcpclientset.NewForConfig(clusterCfg)
+	require.NoError(t, err)
+	return kcpClusterClient
+}
```

Comment:
please not these helpers. They make reading code hard because every package is different. It's not abstracting anything, just do:
```golang
kcpClusterClient, err := kcpclientset.NewForConfig(kcpclienthelper.ConfigWithCluster(config, clusterName))
require.NoError(t, err)
```